### PR TITLE
Add `getLogs` for eth_tester provider

### PR DESCRIPTION
**Pull request by mhchia**
### What was wrong?
`eth_getLogs` was not implemented in `eth-tester` and its provider before.

### How was it fixed?
Implemented it in eth-tester https://github.com/ethereum/eth-tester/pull/50, and mark it as implemented, expand parameters in the function `get_logs`.

#### Cute Animal Picture

Cute animal picture

### CODE REVIEW

**Comment by mhchia** [comment]
Note: I'm not sure if anything should be set for `eth_getLogs` in the tests in eth_tester provider. I can add them if there are some related settings. sorry for this :)

**Comment by mhchia** [comment]
Note: It seems the integration tests for go-ethereum failed. Will look into them tomorrow.

**Comment by mhchia** [comment]
@pipermerriam @carver 
I finally fixed the test. Could you help me to review the tests?
Thank you so much.

File: web3/utils/module_testing/eth_module.py [hunk 1]
```diff
@@ -442,6 +443,80 @@ def test_eth_newPendingTransactionFilter(self, web3):
...
+
+        # the range includes the block where the log resides in
+        filter_params = {
+            "fromBlock": block_with_txn_with_log['number'],
+            "toBlock": block_with_txn_with_log['number'],
+        }
+        result = web3.eth.getLogs(filter_params)
+        assert len(result) == 1
+        log_entry = result[0]
+        assert_is_emitted_log(log_entry)
```
Comment (carver) [inline web3/utils/module_testing/eth_module.py]: Totally at your discretion: this _might_ be a bit cleaner if the assertion did the length check and log extraction. (something like `assert_contains_log(result)`)

**Comment by pipermerriam** [comment]
@mhchia feel free to merge whenever you're happy with this (or let us know it's ready if you don't currently have merge permissions)

File: web3/utils/module_testing/eth_module.py (hunk 1 above)
Comment (mhchia) [reply web3/utils/module_testing/eth_module.py]: Have changed it. Thank you for the suggestion!

**Comment by mhchia** [comment]
@pipermerriam I did just a small change according to @carver 's suggestion.
I think it is ready to merge. Could you help me to merge it since I don't have the merge permission?
Thank you!

**Pull request merged**